File: kata-solutions/tell-dont-ask/tell-dont-ask-python/src/useCase/OrderApprovalRequest.py
from dataclasses import dataclass


@dataclass
class OrderApprovalRequest:
    _order_id: int
    _approved: bool = False

    def get_order_id(self):
        return self._order_id

    def approve(self):
        self._approved = True

    def is_approved(self):
        return self._approved

File: tdd-katas/tell-dont-ask/tell-dont-ask-typescript/src/useCase/OrderApprovalRequest.ts
export default class OrderApprovalRequest {
    private _orderId: number = 0;
    private _approved: boolean= false;


    get orderId(): number {
        return this._orderId;
    }

    set orderId(value: number) {
        this._orderId = value;
    }

    get approved(): boolean {
        return this._approved;
    }

    set approved(value: boolean) {
        this._approved = value;
    }
}